sketch_3DPrinterGUI/gui.pde

```diff
@@ -211,41 +211,42 @@ public void qualitySlider_change(GSlider source, GEvent event) { //_CODE_:infill
 
 //X Textfield Change
 public void xAreaTextfield_change(GTextField source, GEvent event) { //_CODE_:xTextBox:544724:
-  xArea = Integer.parseInt(xTextBox.getText());
-
-  // This is throwing a null pointer exception and crashing atm
-  if (xArea < 1)
-    xTextBox.setText("1");
-  if (xArea > 200)
-    xTextBox.setText("200");
-
-  println("X Area = " + xArea);
+    if (event == GEvent.LOST_FOCUS) {
```

Comment:
The XYZ TextBox does change in the display but the actual variable is still set to the out of range value. Perhaps add the `parseInt()` again inside the if statements and under the `setText()`.